xbmc/xbmc
# SPDX license identifier

## Description
Clean up the mess with copyright notices and replace with an SPDX license identifier. It allows better overview of licenses, for both humans and compliance software. Done following the example of the Linux Kernel folks.

## Motivation and Context
Modernization. Replaces https://github.com/xbmc/xbmc/pull/13927

## How Has This Been Done/Tested?
1. Run [scancode](https://github.com/nexB/scancode-toolkit) on master branch `./scancode -l -n 16 ../kodi --html ../kodi_licenses.html`.
2. Identify all licenses spewed by it.
3. Group files with same license.
4. Confirm group by group, file by file if detected license is correct. Many are false positives because an expression or phrase triggers them. That's the case for **all detected *gpl-1.0-plus***, triggered by `http://www.gnu.org/copyleft/gpl.html`.
5. If license is not clear (missing text, ambiguous interpretation, etc.) try to confirm with original author/sources. That was the case with `xbmc/platform/win32/dxerr.h` and `xbmc/platform/win32/dxerr.cpp` that looked like **MS-PL** licensed files but were indeed **MIT** licensed.
6. After initial assessment, pick a license group and change notice.
7. Loop.
8. Finally, take care of the biggest group, in our case **GPL-2.0-or-later**. Patterns are easy to find and `sed` is a very good friend.
9. Select a random sample of files and have a look to see if everything is OK.

## Current licenses under `xbmc/` (as seen on [SPDX licenses list](https://spdx.org/licenses/)):

| License                                                                   | **[FSF Free/Libre](https://www.fsf.org/licensing)**     | **[OSI Approved](https://opensource.org/licenses/alphabetical)**       | **[GPLv2 Compat](https://www.gnu.org/licenses/license-list.html)**                                                                         |
| ------------------------------------------------------------------------- | :----------------: | :----------------: | :----------------------------------------------------------------------------------: |
| **[BSD-3-Clause](https://spdx.org/licenses/BSD-3-Clause.html)**           | :heavy_check_mark: | :heavy_check_mark: | **[:heavy_check_mark:](https://www.gnu.org/licenses/license-list.html#ModifiedBSD)** |
| **[BSD-4-Clause](https://spdx.org/licenses/BSD-4-Clause.html)**           | :heavy_check_mark: | :x:                | **[:x:](https://www.gnu.org/licenses/license-list.html#OriginalBSD)**      |
| **[GPL-2.0-only](https://spdx.org/licenses/GPL-2.0-only.html)**           | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:                                                                   |
| **[GPL-2.0-or-later](https://spdx.org/licenses/GPL-2.0-or-later.html)**   | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:                                                                   |
| **[LGPL-2.1-or-later](https://spdx.org/licenses/LGPL-2.1-or-later.html)** | :heavy_check_mark: | :heavy_check_mark: | **[:heavy_check_mark:](https://www.gnu.org/licenses/license-list.html#LGPLv2.1)**    |
| **[MIT](https://spdx.org/licenses/MIT.html)**                             | :heavy_check_mark: | :heavy_check_mark: | **[:heavy_check_mark:](https://www.gnu.org/licenses/license-list.html#Expat)**       |
| **[Unlicense (Public Domain)](https://spdx.org/licenses/Unlicense.html)** | :heavy_check_mark: | :question:         | **[:heavy_check_mark:](https://www.gnu.org/licenses/license-list.html#PublicDomain)**        |

Last two commits uniformize our copyright line. Instead of a gazillion variations, we now have:
```
 *  Copyright (C) 2002-2018 Team Kodi
 *  Copyright (C) 2003-2018 Team Kodi
 *  Copyright (C) 2005-2018 Team Kodi
 *  Copyright (C) 2006-2018 Team Kodi
 *  Copyright (C) 2007-2018 Team Kodi
 *  Copyright (C) 2008-2018 Team Kodi
 *  Copyright (C) 2009-2018 Team Kodi
 *  Copyright (C) 2010-2018 Team Kodi
 *  Copyright (C) 2011-2018 Team Kodi
 *  Copyright (C) 2012-2018 Team Kodi
 *  Copyright (C) 2013-2018 Team Kodi
 *  Copyright (C) 2014-2018 Team Kodi
 *  Copyright (C) 2015-2018 Team Kodi
 *  Copyright (C) 2016-2018 Team Kodi
 *  Copyright (C) 2017-2018 Team Kodi
 *  Copyright (C) 2018 Team Kodi
```

Third-party copyright lines were not touched.


## Types of change
